contracts/utility/MathEx.sol

```diff
@@ -158,4 +159,104 @@ library MathEx {
     function roundDivUnsafe(uint256 n, uint256 d) internal pure returns (uint256) {
         return (n + d / 2) / d;
     }
+
+    /**
+      * @dev Compute the largest integer smaller than or equal to `x * y / z`
+    */
+    function mulDivF(uint256 x, uint256 y, uint256 z) internal pure returns (uint256) {
```

Comment:
Could you add natspec for the input parameters and the return value? It's also relevant for some of the methods below.

Reply:
1. It's an internal function, so it won't appear anywhere in our official documentation.
2. What exactly do you want me to say about the input parameters?
3. What exactly do you want me to say about the return value which isn't already mentioned in the 'dev' description?

Reply:
I generally agree, it's just that we currently document everything and then it's harder to refactor (e.g., if an undocumented function becomes public or if we have a new public function with the same parameter, etc.)

Reply:
Let's put a pin in it, while this discussion is being resolved.

Reply:
'put a pin' as in...? add dummy natspec for now?

Reply:
I meant that you can keep at as-is for now and we will revisit it later